server/src/main/java/org/apache/druid/indexing/overlord/supervisor/Supervisor.java

```diff
@@ -93,4 +93,9 @@ default Boolean isHealthy()
   LagStats computeLagStats();
 
   int getActiveTaskGroupsCount();
+
+  /** Handoff the task group with id=taskGroupId the next time the supervisor runs regardless of task run time*/
+  default void handoffTaskGroupEarly(int taskGroupId)
+  {
+  }
```

Comment:
```suggestion
  default void handoffTaskGroupEarly(int taskGroupId) throws DruidException
  {
    throw DruidException
            .forPersona(DruidException.Persona.ADMIN)
            .ofCategory(DruidException.Category.UNSUPPORTED)
            .build(...);
  }
```

I think the default implementation should indicate in some way that this is not supported. I don't have a strong opinion on if it needs to be via an exception or some other mechanism.